src/libbson/src/bson/bson-string.c

```diff
@@ -31,6 +31,47 @@
 #include <string.h>
 #endif
 
+// `bson_next_power_of_two_u32` returns 0 on overflow.
+static BSON_INLINE uint32_t
+bson_next_power_of_two_u32 (uint32_t v)
+{
+   BSON_ASSERT (v > 0);
+
+   v--;
```

Comment:
```suggestion
   // https://graphics.stanford.edu/%7Eseander/bithacks.html#RoundUpPowerOf2
   v--;
```
Recommend adding a reference to a source explaining this algorithm.